### DISCUSSION TITLE
Adding new sensors

### DISCUSSION THREAD

**Discussion by paoloelle** [turtlebot/turtlebot4#475] [Q&A] [answered]
Hi all,
I'm trying to add three light sensors to the TurtleBot4. These sensors are the [Adafruit TCS34725 RGB color senors](https://www.adafruit.com/product/1334l) that uses the I2C communication protocol. As far as I've seen from the TurtleBot4 docs, in the section [User Interface PCBA](https://turtlebot.github.io/turtlebot4-user-manual/electrical/pcba.html) I should be able to add multiple of them since various GPIOs are available (2,3 8,9, 10,11) that are configurable as SDA, SCL. The problem I'm facing is that I can detect the sensor and read the data if I connect it to the GPIO 2 and GPIO3 but I don't know how to read the data from the other sensors.

**Comment by slowrunner** [comment]
(Old Post - @paoloelle still around?)

The Raspberry Pi defaults to having one I2C bus and the Adafruit RGB color sensor you linked appears to have a single I2C address 0x29 available, so with normal configuration only one of these sensors may be connected to the default I2C bus.

Configuring a second and third I2C bus with user access and providing the 3.3v to 5v I2C bus level shift/output protection is complicated for sure.

How is your project going?

**Comment by paoloelle** [comment, marked as answer]
HI @slowrunner I found a way to do that.You can take a look at my answer on Robotics Stack Exchange [here](https://robotics.stackexchange.com/a/114090/38519).
